New copy to clipboard shortcut feature and footer icon

DSGDSR opened the pull request:
Hi! I just created this PR adding a new shortcut for copying the selected node. I leave some images here... tried to follow svg structure you have in the project 😃 

<img width="1101" alt="Captura de pantalla 2022-04-27 a las 14 35 50" src="https://user-images.githubusercontent.com/9610219/165519512-d4cf3c21-6ab6-4735-ae5f-98ee40ee24e2.png">
<img width="1101" alt="Captura de pantalla 2022-04-27 a las 14 35 56" src="https://user-images.githubusercontent.com/9610219/165519517-8ac7119d-b265-454c-b292-640749aa7055.png">

Few days using JSON Hero t handle big json files and feels quite comfortable... awesome app
Let me know how it looks, in case you want me to add/edit something let me know! Thanks

ericallam commented:
Hi @DSGDSR, thanks for the contribution! I'll be taking a look at this in more detail on Friday. Glad you like JSON Hero and thanks for being our first outside PR 💯

ericallam commented:
This is a great addition! Thanks for the PR. I do have a couple of things that I think will need to be done before this can be merged:

- [ ] Choose a different keyboard shortcut. It's a little dangerous overriding the default copy shortcut as it breaks copying anything else on the page when selecting some text and hitting `CMD/CTRL+C`
- [ ] Don't `JSON.stringify` strings to prevent the extraneous `""` from being copied to the clipboard.  You can do that by checking for `selectedInfo?.name === "string"`
- [ ] Ideally this would also work in the Tree View but currently it only works in the Column View. Not a dealbreaker for getting this PR merged as we could add support for the Tree View later without much problem.

DSGDSR commented:
Hi @ericallam, just updated the branch with the changes:

- Using `shift+c` now as shortcut, also thought about using `cmd+u`, but this would work for Windows too.
- Isolated component for this specific shortcut as it is used in the TreeView too, this way it was easily implemented instead of adding it to `useVirtualTree` hook as the rest of shortcuts.
- Checking the selected node name, in case it is an string the value would be copied to the clipboard without using `JSON.stringify`
- Modified the SVG accordingly:
<img width="558" alt="Captura de pantalla 2022-05-02 a las 12 32 17" src="https://user-images.githubusercontent.com/9610219/166221758-079b3be9-353b-42d4-8fb0-3960fd8a8577.png">

ericallam commented on app/components/Icons/CopyShortcutIcon.tsx:
```diff
@@ -0,0 +1,23 @@
+      <rect x="16" width="14" height="14" rx="1.53846" fill="currentColor" />
+      <path
+        d="M5.64,10.22H8.25V7a.39.39,0,0,1,.38-.39H10l-3-3L4,6.65H5.26A.39.39,0,0,1,5.64,7v3.18Zm3,.78H5.26a.38.38,0,0,1-.39-.39V7.43H3.11a.39.39,0,0,1-.28-.66L6.72,2.86a.39.39,0,0,1,.55,0l3.88,3.91a.38.38,0,0,1-.27.66H9v3.18a.38.38,0,0,1-.39.39Z"
+        stroke="#0f172a" stroke-width="0.35px" fill="#0f172a"
```
This attribute should be `strokeWidth`, not `stroke-width`

DSGDSR replied:
Comment resolved 😺, thanks for the support and for accepting my proposal. Will be back if any feature comes to my mind while using jsonhero!

ericallam commented:
This PR will be ready to merge once the above comment is resolved. Great work, such an awesome feature 👍
